### DISCUSSION TITLE
"CreateBucket operation: Access Denied..." when calling Session.default_bucket()

### DISCUSSION THREAD

**Discussion by kianho** [aws/sagemaker-python-sdk#2396] [Help]
**Describe the bug**
Calling `default_bucket()` on a `sagemaker.session.Session` object, when specifying a custom `default_bucket`, raises:

* `ClientError: An error occurred (AccessDenied) when calling the CreateBucket operation: Access Denied`

when the default bucket already exists

possibly related to https://github.com/aws/sagemaker-python-sdk/issues/1174

**To reproduce**
```python
bucket = "bucket-that-exists"
boto_session = boto3.session.Session(region_name="ap-southeast-2")
sm_session = sagemaker.Session(boto_session, default_bucket=bucket)

sm_session.default_bucket()
```
* I added a command to L387 of `src/sagemaker/session.py` to display the number of keys in the bucket to verify that the intended default bucket is accessed:

![image](https://user-images.githubusercontent.com/783366/94792220-fd959980-041b-11eb-9fe4-88d57a706336.png)

* `bucket.creation_date` returns `None` even when the bucket exists
* could be tricky to reproduce, the execution role has limited permissions (can only list, upload/download, to/from this bucket), for example `aws s3 ls` is denied, let me know if there is any more info you need

![image](https://user-images.githubusercontent.com/783366/94793341-ac86a500-041d-11eb-9776-78d002086967.png)

**Expected behavior**
`sm_session.default_bucket()` should return the same default bucket name specified when constructing the sagemaker session object.

**Screenshots or logs**
If applicable, add screenshots or logs to help explain your problem.
![image](https://user-images.githubusercontent.com/783366/94790753-11d89700-041a-11eb-89f1-6bcb7d35899b.png)

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: 2.12.0 (boto3 1.14.60)
- **Python version**: Python 3.6.10 :: Anaconda, Inc.

**Comment by ajaykarpur** [comment]
Does your role have access to the `bucket`?

It looks like https://github.com/aws/sagemaker-python-sdk/issues/1174 was addressed in https://github.com/aws/sagemaker-python-sdk/pull/1176.
